src/core/server/metrics/metrics_service.test.ts

```diff
@@ -57,37 +57,46 @@ describe('MetricsService', () => {
       expect(setInterval).toHaveBeenCalledWith(expect.any(Function), testInterval);
     });
 
-    it('emits the metrics at start', async () => {
+    it('collects the metrics at every interval', async () => {
       mockOpsCollector.collect.mockResolvedValue(dummyMetrics);
 
-      const { getOpsMetrics$ } = await metricsService.setup({
-        http: httpMock,
-      });
-
+      await metricsService.setup({ http: httpMock });
       await metricsService.start();
 
       expect(mockOpsCollector.collect).toHaveBeenCalledTimes(1);
-      expect(
-        await getOpsMetrics$()
-          .pipe(take(1))
-          .toPromise()
-      ).toEqual(dummyMetrics);
+
+      jest.advanceTimersByTime(testInterval);
+      expect(mockOpsCollector.collect).toHaveBeenCalledTimes(2);
+
+      jest.advanceTimersByTime(testInterval);
+      expect(mockOpsCollector.collect).toHaveBeenCalledTimes(3);
     });
 
-    it('collects the metrics at every interval', async () => {
+    it('resets the collector after each collection', async () => {
       mockOpsCollector.collect.mockResolvedValue(dummyMetrics);
 
-      await metricsService.setup({ http: httpMock });
-
+      const { getOpsMetrics$ } = await metricsService.setup({ http: httpMock });
       await metricsService.start();
 
+      const waitForNextEmission = () =>
+        getOpsMetrics$()
+          .pipe(take(1))
+          .toPromise();
+
       expect(mockOpsCollector.collect).toHaveBeenCalledTimes(1);
+      expect(mockOpsCollector.reset).toHaveBeenCalledTimes(1);
 
+      let nextEmission = waitForNextEmission();
       jest.advanceTimersByTime(testInterval);
+      await nextEmission;
```

Comment:
```ts
  private async refreshMetrics() {
    this.logger.debug('Refreshing metrics');
    const metrics = await this.metricsCollector!.collect();
    this.metricsCollector!.reset();
    this.metrics$.next(metrics);
  }
```

`this.metricsCollector!.collect()` is the first thing called in the interval handler, so waiting for `advanceTimersByTime` is sufficient, however as the `reset` call is performed after an async operation, the `reset` was not called on time for the test, so I had to actually wait for the observable emission

Reply:
does it make sense to add a comment in test?

Reply:
Probably. Will do